src/LoggerConfigurator.cpp

```diff
@@ -0,0 +1,180 @@
+
+// LoggerConfigurator - reading strings for configuring the Logger
+//
+// Author  : Klaus-Dieter Gundermann
+// Created : 30.06.2020
+//
+// For all support, instructions and copyright go to:
+// http://e2guardian.org/
+// Released under the GPL v2, with the OpenSSL exception described in the README file.
+
+#ifdef HAVE_CONFIG_H
+#include "e2config.h"
+#endif
+
+#include <algorithm>
+// #include <string>
+#include "LoggerConfigurator.hpp"
+#include "String.hpp"
+
+// -------------------------------------------------------------
+// --- Constructor
+// -------------------------------------------------------------
+LoggerConfigurator::LoggerConfigurator(Logger *logger) {
+    _logger = logger;
+};
+
+LoggerConfigurator::~LoggerConfigurator() {
+    _logger = NULL;
+};
+
+const String LoggerConfigurator::Prefix("log_");
+
+// -------------------------------------------------------------
+// --- Public Functions
+// -------------------------------------------------------------
+
+bool LoggerConfigurator::configure(LoggerSource source, const std::string option) {
+    String line(option);
+    line.removeChar(' ');
+    std::string destination;
+    String filename;
+    if(line.contains(":")) {
+        destination = line.before(":");
+        filename = line.after(":");
+        filename.removeChar('\'');
+    } else {
+        destination = line;
+    }
+
+    LoggerDestination dst = _logger->string2dest(destination);
+
+    DEBUG_config("LoggerConfig:", " source:", _logger->source2string(source), " destination:", destination, " filename:", filename);
+
+    return _logger->setLogOutput(source, dst, filename);
+}
+
+bool LoggerConfigurator::configure(const std::string option) {
+    String line(option);
+    line.removeChar(' ');
+    if (!line.startsWith(Prefix)) return false;
+
+    size_t pos1 = line.find("=", 0);
+    if (pos1 == std::string::npos) return false;
+
+    std::string source = line.substr(Prefix.size(), pos1 - Prefix.size());
+
+    std::string reduced_option;
+
+    reduced_option = line.substr(pos1 + 1);
+    LoggerSource src = _logger->string2source(source);
+
+    return configure(src, reduced_option);
+};
+
+bool LoggerConfigurator::debuglevel(const std::string option) {
```

Comment:
Codacy found an issue: [Function parameter 'option' should be passed by const reference.](https://app.codacy.com/manual/numsys/e2guardian/pullRequest?prid=6238852)